File: sample/src/main/java/com/smartnsoft/smartrecyclerviewsample/ReorderedRecyclerActivity.java
package com.smartnsoft.smartrecyclerviewsample;

import com.smartnsoft.recyclerview.adapter.SmartRecyclerAdapter;
import com.smartnsoft.recyclerview.adapter.SmartReorderAdapter;

/**
 * An activity containing a simple way to use a ReorderRecyclerView with a SmartReorderAdapter
 *
 * @author Adrien Vitti
 * @since 2018.01.24
 */
public final class ReorderedRecyclerActivity
    extends AbstractRecyclerActivity
{

  @Override
  protected int getLayoutId()
  {
    return R.layout.reordered_recycler_activity;
  }

  @Override
  protected SmartRecyclerAdapter createRecyclerAdapter()
  {
    final SmartReorderAdapter smartReorderAdapter = new SmartReorderAdapter(this);
    smartReorderAdapter.setHasStableIds(true);
    return smartReorderAdapter;
  }

}

File: sample/src/main/java/com/smartnsoft/smartrecyclerviewsample/ReorderedRecyclerActivity.kt
package com.smartnsoft.smartrecyclerviewsample

import com.smartnsoft.recyclerview.adapter.SmartRecyclerAdapter

/**
 * An activity containing a simple way to use a ReorderRecyclerView
 *
 * @author Adrien Vitti
 * @since 2019.07.15
 */
class ReorderedRecyclerActivity : AbstractRecyclerActivity()
{

  override fun getLayoutId(): Int
  {
    return R.layout.reordered_recycler_activity
  }

  override fun createRecyclerAdapter(): SmartRecyclerAdapter
  {
    return SmartRecyclerAdapter(this, true).apply { setHasStableIds(true) }
  }

  override fun shouldActivateTextReordering(): Boolean
  {
    return true
  }

}